lib/kernel.coffee

```diff
@@ -83,8 +87,11 @@ class Kernel
 
     # onResults is a callback that may be called multiple times
     # as results come in from the kernel
-    execute: (code, onResults) ->
-        requestId = uuid.v4()
+    execute: (code, isWatch, onResults) ->
+        if isWatch
+            requestId = "watch_" + uuid.v4()
+        else
+            requestId = "execute_" + uuid.v4()
 
```

Comment:
This is your own opinion on style here, but I'd make wrapper functions here that call an underlying `_execute` with the `requestID`, solely so execute is about the the code:

```
_execute: (code, requestId, onResults) ->
   ...
execute: (code, onResults) ->
  requestId = "execute_" + uuid.v4()
executeWatch(code, onResults) ->
  requestId = "watch_" + uuid.v4()
```